src/log4net/Repository/Hierarchy/Hierarchy.cs

```diff
@@ -252,12 +249,7 @@ public ILoggerFactory LoggerFactory
     /// </remarks>
     public override ILogger? Exists(string name)
     {
-      if (name is null)
-      {
-        throw new ArgumentNullException(nameof(name));
-      }
-
-      m_ht.TryGetValue(new LoggerKey(name), out object? o);
+      loggers.TryGetValue(new(name.EnsureNotNull()), out object? o);
```

Comment:
I would have kept `name.EnsureNotNull()` on its own line at the start.

Reply:
inlining EnsureNotNull() is one of the reasons this method exists ;-)